mypy/main.py

```diff
@@ -516,6 +517,15 @@ def add_invertible_flag(flag: str,
     if options.quick_and_dirty:
         options.incremental = True
 
+    if options.python_executable:
+        check = subprocess.check_output([options.python_executable, '-V'],
+                                        stderr=subprocess.STDOUT).decode('UTF-8')
+        assert check.startswith('Python'), \
+            "Mypy could not use the Python executable: {}".format(options.python_executable)
+        ver = re.fullmatch(r'Python (\d)\.(\d)\.\d\s*', check)
+        if ver:
+            options.python_version = (int(v) for v in ver.groups())
```

Comment:
Hm, this seems backwards. I'd much rather specify `--python-version 3.4` than `--python-executable /usr/local/bin/python3.4`. At the very least it shouldn't silently overwrite `python_version` when explicitly given.

Reply:
I suppose I can not have this overwrite `python_version`, but finding a python executable based on a given version is a somewhat intractable problem. Unless I'm missing something there isn't a way to query and be informed about the install location of Pythons in a cross platform manner.

Reply:
Sadly you are right, this can't be solved cross-platform. On UNIX platforms, searching `$PATH` for a binary named `pythonX.Y` should work. But on Windows I think the binary is always named `Python.exe`?

In any case, mypy's current rule is that the target Python version is specified using `-python-version=X.Y` and always defaults to 3.6, regardless of what Python version is used to *run* mypy. I'd rather not change this contract mypy's users.

Maybe we can make the default be something like *best effort* which means on UNIX we search `$PATH` for `pythonX.Y` and on Windows we verify that `python.exe` is the right version (using `-V`). In either case if the binary can't be found or doesn't match `--python-version` you can print a warning that installed packages won't be found.

(Hm... Is there really no common convention for where different versions are installed on Windows? Sadly I don't have a Windows system around to check this. But I found a few related StackOverflow questions -- maybe this one is most useful? https://stackoverflow.com/questions/15912063/how-do-i-run-python-2-and-3-in-windows-7/17245543#17245543 )

Reply:
I can work on a best-effort based approach. I believe we should be able to use the `py` launched on Windows (which makes this much easier, since we can do `pythonX.Y` on Unix, and `py -X.Y` on Windows).